app/src/org/commcare/dalvik/activities/SingleAppManagerActivity.java

```diff
@@ -0,0 +1,285 @@
+        }
+        
+        //refresh validate button
+        Button validateButton = (Button) findViewById(R.id.verify_button);
+        if (appRecord.resourcesValidated()) {
+            validateButton.setVisibility(View.INVISIBLE);
+        } else {
+            validateButton.setVisibility(View.VISIBLE);
+        }
+    }
+    
+    @Override
+    protected void onActivityResult(int requestCode, int resultCode, Intent intent) {
+        switch (requestCode) {
+        case CommCareHomeActivity.UPGRADE_APP:
+            if(resultCode == RESULT_CANCELED) {
+                Toast.makeText(this, "Your update did not complete", Toast.LENGTH_LONG).show();
+            } else if(resultCode == RESULT_OK) {
+                //Set flag that we should autoupdate on next login
+                SharedPreferences preferences = CommCareApplication._().getCurrentApp().getAppPreferences();
+                preferences.edit().putBoolean(CommCarePreferences.AUTO_TRIGGER_UPDATE,true);
+            }
+            break;
+        case CommCareHomeActivity.MISSING_MEDIA_ACTIVITY:
+            refresh();
+            if (resultCode == RESULT_CANCELED) {
+                AlertDialog.Builder builder = new AlertDialog.Builder(this);
+                builder.setTitle("Media Not Verified");
+                builder.setMessage(R.string.skipped_verification_warning_2)
+                    .setPositiveButton("OK", new DialogInterface.OnClickListener() {
+
+                            @Override
+                            public void onClick(DialogInterface dialog,
+                                    int which) {
+                                dialog.dismiss();
+                            }
+                            
+                        });
+                dialog = builder.create();
+                dialog.show();
+            }
+            else if (resultCode == RESULT_OK) {
+                Toast.makeText(this, "Media Validated!", Toast.LENGTH_LONG).show();
+            }
+            break;
+        case CommCareHomeActivity.RESTART_APP:
+            if (dialog != null) {
+                dialog.dismiss();
+            }
+            Intent i = new Intent(getApplicationContext(), AppManagerActivity.class);
+            startActivity(i);
+        }
+    }
+    
+    /** Uninstalls the selected app **/
+    public void uninstall() {
+        CommCareApplication._().logout();
+        CommCareApplication._().initializeAppResources(new CommCareApp(appRecord));
+        CommCareApp app = CommCareApplication._().getCurrentApp();
+        
+        //1) Teardown the sandbox for this app
+        app.teardownSandbox();
+        //2) Delete all the user databases associated with this app
+        SqlStorage<UserKeyRecord> userDatabase = CommCareApplication._().getAppStorage(UserKeyRecord.class);
+        for (UserKeyRecord user : userDatabase) {
+            this.getDatabasePath(CommCareUserOpenHelper.getDbName(user.getUuid())).delete();
```

Comment:
Is there a cleaner/less file system specific way to do the DB delete than manually identifying the file?